lib/llm/response/completion.rb

```diff
@@ -6,37 +6,47 @@ class Response::Completion < Response
     # @return [String]
     #   Returns the model name used for the completion
     def model
-      @provider.__send__ :completion_model, raw
+      parsed[:model]
     end
 
     ##
     # @return [Array<LLM::Message>]
     #  Returns an array of messages
     def choices
-      @provider.__send__ :completion_choices, raw
+      parsed[:choices]
     end
 
     ##
     # @return [Integer]
     #   Returns the count of prompt tokens
     def prompt_tokens
-      @provider.__send__ :completion_prompt_tokens, raw
+      parsed[:prompt_tokens]
     end
 
     ##
     # @return [Integer]
     #   Returns the count of completion tokens
     def completion_tokens
-      @provider.__send__ :completion_completion_tokens, raw
+      parsed[:completion_tokens]
     end
 
     ##
     # @return [Integer]
     #   Returns the total count of tokens
     def total_tokens
       @provider.respond_to?(:completion_total_tokens) ?
-        @provider.__send__(:completion_total_tokens, raw) :
+        parsed[:total_tokens] :
         prompt_tokens + completion_tokens
     end
+
+    private
+
+    ##
+    # @private
```

Comment:
No need to add `@private` here. Yardoc will use Ruby's visibility rules, and this method is already private.